### PR TITLE
Smaller change to switch from Java 7 to Java 8 as minimum Java level

### PR DESCRIPTION
**Pull request by jhanders34**
- Update current Java 6 components to be Java 7.
- Update the default settings to have a Required-Capability to be Java 7.  It should have been the default for a long time now.
- Update test case that wouldn't compile with Java 8.
- Update checking for Java version to be a Java 8 check now instead of Java 7.

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

File: dev/cnf/resources/bnd/bundle.props [hunk 1]
```diff
@@ -38,7 +38,7 @@ Bundle-Version: ${bFullVersion}.${if;${driver;gradle};${version.qualifier};eclip
 -consumer-policy :    ${range;[==,+)}
 
 # Require a 1.6 JDK (at least): see section 3.4.1 of OSGi 4.3 core spec
```
Comment (aguibert) [inline dev/cnf/resources/bnd/bundle.props]: comments should be updated here and in rejar.props because they explicitly refer to JDK 1.6

Comment (jhanders34) [reply dev/cnf/resources/bnd/bundle.props]: I talked with Andy.  As I expressed in the other PR, the comment is referring specifically to OSGi requirements and we don't want this default ever set to something lower than 6.  7 is at least 6 which is what the comment is stating.

**Comment by jhanders34** [comment]
#build

**Comment by skasund** [comment]
L2 message review completed

**Pull request merged**